Frontend/ui-customer/src/Screens/ZooTicketBookingFlow/TicketBookingScreen/View.purs

```diff
@@ -293,244 +365,696 @@ feeBreakUpView state push services =
+individualTicketBHView push state valBH ticket =
+  PrestoAnim.animationSet [
+    Anim.translateInYAnim translateYAnimConfig { duration = 3000 , fromY = -10 , toY = 0}
+  ] $ 
+  linearLayout
+  [ height WRAP_CONTENT
+  , width MATCH_PARENT
+  , orientation VERTICAL
+  ] $ [
+  ] 
+    <> (case valBH of 
+          Nothing -> []
+          Just bh -> if DA.length bh.categories > 1 then [selectDestinationView push state bh.categories] else (map (incrementDecrementView push state) (convertBHToPeopleCategory bh.categories)))
+    <> (if DA.length ticket.slot > 0 then [timeSlotView push state ticket.ticketID ticket.selectedBHid ticket.slot] else [])
+  where 
+    convertBHToPeopleCategory categories = do
+      let mbCurrentCategory = DA.head categories
+      case mbCurrentCategory of
+        Nothing -> []
+        Just category -> category.peopleCategories
+
+selectDestinationView :: forall w . (Action -> Effect Unit) -> ST.TicketBookingScreenState -> Array ST.TicketCategoriesOptionData -> PrestoDOM (Effect Unit) w
+selectDestinationView push state categories =
+  linearLayout
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  , orientation VERTICAL
+  ] [ linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , orientation VERTICAL
+      ] (map (incrementDecrementView push state) (convertBHToPeopleCategory filteredCategories))
+    , textView $
+      [ text "Select your ticket category"
+      , color Color.black800
+      , margin $ MarginVertical 20 8
+      ] <> FontStyle.subHeading1 TypoGraphy
+    , linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , orientation VERTICAL
+      ] (map (selectDestinationViewPill push state) categories)
+  ]
+  where filteredCategories = DA.filter (\category -> category.isSelected) categories
+        convertBHToPeopleCategory categories = do
+          let mbCurrentCategory = DA.head categories
+          case mbCurrentCategory of
+            Nothing -> []
+            Just category -> category.peopleCategories
+
+timeSlotView :: forall w . (Action -> Effect Unit) -> ST.TicketBookingScreenState -> String -> Maybe String -> Array ST.SlotInterval -> PrestoDOM (Effect Unit) w
+timeSlotView push state ticketID selectedBHid slots = 
+  let filteredSlots = getFilteredSlots slots state
+  in
+  linearLayout
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  , orientation VERTICAL
+  , visibility $ if DA.length filteredSlots > 0 then VISIBLE else GONE
+  ] [ textView $
+      [ text "Time Slot"
+      , color Color.black800
+      , margin $ MarginVertical 20 8
+      ] <> FontStyle.subHeading1 TypoGraphy
+    , linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , orientation VERTICAL
+      ] ( map (timeSlotRowView push ticketID selectedBHid) (convertTimeSlotsToGroupedArray filteredSlots))
+  ]
+  where
+    getFilteredSlots slots state = do
```

Comment:
do block is not required as there is not monadic operations

Reply:
done